src/engine/strat_engine/pool.rs

```diff
@@ -151,9 +151,16 @@ impl Pool for StratPool {
         // TODO: Roll back on filesystem initialization failure.
         let dm = DM::new()?;
         let mut result = Vec::new();
-        for name in names {
-            let fs_uuid = self.thin_pool.create_filesystem(name, &dm, None)?;
-            result.push((*name, fs_uuid));
+        let mut names_iter = names.into_iter();
+        if let Some(first_name) = names_iter.next() {
```

Comment:
This whole approach seems a little circumscribed to me. Why can't each stratisd pool just keep around a secret, empty filesystem, which it snapshots whenever any request for a new filesystem comes in? Here this is an optimization which is not terribly useful if the user just requests one filesystem at a time.

How much faster is snapshotting than making a whole new filesystem, anyway?

Reply:
Yes absolutely. This is a stepping stone to getting to that implementation, which has additional complexities and larger code changes that it would need.

> How much faster is snapshotting than making a whole new filesystem, anyway?

It's not! Very strange! See #694.